src/entity/EnemyShipDiver.java

```diff
@@ -0,0 +1,81 @@
+package entity;
+
+import engine.Cooldown;
+import engine.Core;
+import engine.DrawManager;
+import engine.GameState;
+
+import java.awt.*;
+
+public class EnemyShipDiver extends EnemyShip {
+
+    private static final int POINTS = 50;
+    private final int DIVE_INTERVAL = 4000;
+    private final int DIVE_VARIANCE = -1000;
+    public static final int SPEED_X = 10;
+    public static final int SPEED_DIVE = 3;
+    private Cooldown diveCooldown;
+    private int state;
+
+
+    /**
+     * Creates a diver ship with the given x and y positions
+     * @param positionX
+     * @param positionY
+     */
+    public EnemyShipDiver(final int positionX, final int positionY, final GameState gameState) {
+        super(positionX, positionY, DrawManager.SpriteType.EnemyShipE1, gameState);
+        this.diveCooldown = Core.getVariableCooldown(DIVE_INTERVAL, DIVE_VARIANCE);
+        this.diveCooldown.reset();
+        this.pointValue = POINTS;
+        this.state = (int)Math.round(Math.random());
+    }
+
+    /**
+     * Updates attributes, mainly used for animation purposes.
+     */
+    public final void update() {
+        if((this.state / 10) % 2 == 0 && this.state != 2) {
+            this.color = Color.YELLOW;
+        } else {
+            this.color = Color.RED;
+        }
+
+        if (this.animationCooldown.checkFinished()) {
+            this.animationCooldown.reset();
+
+            switch (this.spriteType) {
+                case EnemyShipE1:
+                    this.spriteType = DrawManager.SpriteType.EnemyShipE2;
+                    break;
+                case EnemyShipE2:
+                    this.spriteType = DrawManager.SpriteType.EnemyShipE1;
```

Comment:
This looks good to me, but would it not be better to use the `EnemyShipF1` sprite instead of `EnemyShipE1` ?

Reply:
That could work -- I chose E1 because it F1 was noted as an elite enemy in the DrawManager. I can change it to F1